blocks/library/gallery/gallery-image.js

```diff
@@ -1,5 +1,25 @@
 
 export default function GalleryImage( props ) {
+	if ( props.linkto === 'media' ) {
+		return (
+			<figure className="blocks-gallery-image">
+				<a href={ props.img.url }>
+					<img src={ props.img.url } alt={ props.img.alt } />
+				</a>
+			</figure>
+		);
+	}
+
+	if ( props.linkto === 'attachment' && !! props.img.link ) {
```

Comment:
It'd be nice to not duplicate the markup for links and do this instead:
https://github.com/WordPress/gutenberg/blob/master/blocks/library/image/index.js#L200